### PR TITLE
Bugfix/7873 time conductor input validation

### LINKED ISSUE #7873: Time Conductor Fixed Time inputs are using "punish early" validation
Manually editing Start and End times in the Time Conductor popup now fire validation on keystroke, rather than field blur. This has two bad UX results:
#### Time Field
Editing the time immediately shows an 'invalid date' error in the `date` input as the user deletes a character in the `time` inputs, and worst of all, jumps the input cursor into the `date` input, which is actually well-formed. The user will almost always type a bad number into the `date` field as a result, making it harder to correct the situation.

**Initial state**
Screenshot 2024-10-08 at 11 50 50 AM

**Delete the '5' intending to change it to a '0'** Note that the validation message points to the wrong input.
Screenshot 2024-10-08 at 11 51 01 AM

**Input cursor has jumped into the `date` input. User's finger in slow-motion types a '0'**
Screenshot 2024-10-08 at 12 07 55 PM
#### Date Field
Editing the date immediately shows an 'invalid date' error as the user deletes a character prior to typing a new one.
**Delete the '8' intending to change it.**
Screenshot 2024-10-08 at 11 54 45 AM

#### Expected vs Current Behavior
This has become the "punish early" UI model of form validation, which is bad. 

- We should only be validating inputs when the user blurs out of an input in the popup, and we _definitely_ should not be jumping the user's input cursor into a different field than the one they were working in. 
- The 'Invalid' message should point to the actual bad input.

#### Steps to Reproduce

Follow steps above.

#### Environment

* Open MCT Version:  4.1.0-next / RC10
* Deployment Type:  localhost, server
* OS: any
* Browser: both Chrome and Firefox tested.

#### Impact Check List

- [ ] Data loss or misrepresented data?
- [x] Regression? Did this used to work or has it always been broken?
- [ ] Is there a workaround available?
- [x] Does this impact a critical component?
- [ ] Is this just a visual bug with no functional impact?
- [ ] Does this block the execution of e2e tests?
- [ ] Does this have an impact on Performance?

### PR DESCRIPTION
**Pull request by davetsay**
Closes #7873 

### Describe your changes:
* Checks validation of inputs on `input` event
* Only reports validation of inputs on `change` event, so user is not punished early
* Only reports logical validation errors on submit (limit out of bounds, start bounds greater than end bounds, etc.)

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [ ] Is this a [notable change](../docs/src/process/release.md) that will require a special callout in the release notes? For example, will this break compatibility with existing APIs or projects that consume these plugins?

### Author Checklist

* [x] Changes address original issue?
* [x] Tests included and/or updated with changes?
* [x] Has this been smoke tested?
* [x] Have you associated this PR with a `type:` label? Note: this is not necessarily the same as the original issue.
* [x] Have you associated a milestone with this PR? Note: leave blank if unsure.
* [x] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [ ] Changes appear to address issue?
* [ ] Reviewer has tested changes by following the provided instructions?
* [ ] Changes appear not to be breaking changes?
* [ ] Appropriate automated tests included?
* [ ] Code style and in-line documentation are appropriate?

### CODE REVIEW

File: src/plugins/timeConductor/TimePopupFixed.vue [hunk 1]
```diff
@@ -11,18 +11,19 @@
...
-          v-model="formattedBounds.start"
+          v-model="formattedBounds.startDate"
           class="c-input--datetime"
           type="text"
           autocorrect="off"
           spellcheck="false"
           aria-label="Start date"
-          @input="validateAllBounds('startDate')"
+          @input="onInput('startDate')"
+          @change="onChange('startDate')"
```
Comment (shefalijoshi) [inline src/plugins/timeConductor/TimePopupFixed.vue:21]: We should use something more descriptive than `onInput` and `onChange`.

**Review by shefalijoshi — APPROVED** [review]
Small change but looks solid

File: src/plugins/timeConductor/TimePopupFixed.vue [hunk 2]
```diff
@@ -215,96 +232,98 @@ export default {
...
+    clearAllValidation() {
+      Object.keys(this.inputValidityMap).forEach(this.clearValidation);
+    },
+    clearValidation(refName) {
+      const input = this.getInput(refName);
+
+      input.setCustomValidity('');
+      input.title = '';
+    },
+    onInput(refName) {
```
Comment (akhenry) [inline src/plugins/timeConductor/TimePopupFixed.vue:244]: Please give this a name that describes what it does, not the event that it handles. eg. `clearValidationAndRevalidate`

File: src/plugins/timeConductor/TimePopupFixed.vue [hunk 3]
```diff
@@ -215,96 +232,98 @@ export default {
...
+      this.clearAllValidation();
+      this.validateInput(refName);
+    },
+    onChange(refName) {
```
Comment (akhenry) [inline src/plugins/timeConductor/TimePopupFixed.vue:248]: Is this needed? It's just calling `handleValidationResults`

**Review by akhenry — CHANGES_REQUESTED** [review]

**Comment by akhenry** [comment]
Tested this locally and it looks really good, very nice work. I noticed a _very_ minor niggle that could be fixed in a followup -


https://github.com/user-attachments/assets/c6c713e1-a76e-47e2-82a8-35a724fd8f88

**Comment by davetsay** [comment]
> Tested this locally and it looks really good, very nice work. I noticed a _very_ minor niggle that could be fixed in a followup. Invalid time bounds result in the submit button being disabled which is great. However an invalid combination of dates leaves the submit button _enabled_. If you try and click the enabled submit button it shows a validation error but for consistency it would be great if invalid date bounds also disabled the submit button, like invalid times do
> 
> eg.
> 
>  minor.TC.bug.mov

This is by design. The point is to not punish the user early. Give them a chance to input valid bounds, and if they don't, prevent them from submitting invalid bounds. This comes into play when changing both start and end bounds. The user almost always edits the start bounds first, and then is punished too early if say they are moving the start bounds and end bounds forward in time to a time window that is after the end bounds.

I think this is what the video is showing. If not, we can re-examine.

**Comment by davetsay** [comment]
@akhenry , @shefalijoshi , addressed your comments in this commit, https://github.com/nasa/openmct/pull/7886/commits/8d4d57d0ceb1edfce2fe2ac8835dd305135fb8b5. ready for re-review

**Review by akhenry — APPROVED** [review]

**Comment by akhenry** [comment]
@davetsay  Love it, the validation is subtle and brilliant. For all intents and purposes it's art.

**Pull request merged**